synapse/storage/databases/main/event_federation.py

```diff
@@ -1292,6 +1292,22 @@ def _get_backfill_events(
 
         return event_id_results
 
+    @trace
+    async def record_event_backfill_attempt(self, event_id: str) -> None:
+        await self.db_pool.simple_upsert(
+            table="event_backfill_attempts",
+            keyvalues={"event_id": event_id},
+            values={
+                "event_id": event_id,
+                # TODO: This needs to increment
+                "num_attempts": 1,
```

Comment:
Do we have any common pattern to increment something on `upsert`?

Reply:
@clokep mentioned that there probably isn't a helper for this.

I based my example on the custom `upsert` we use `set_destination_retry_timings`: [`synapse/storage/databases/main/transactions.py#L207-L330`](https://github.com/matrix-org/synapse/blob/978666a088aacf467d3d265ff3b7b09c0d19733d/synapse/storage/databases/main/transactions.py#L207-L330)

Reply:
https://github.com/matrix-org/synapse/pull/13589/commits/f9119d0a9ea8f011babe7f6851a4033d3294f018